app/parse.py

```diff
@@ -15,9 +21,83 @@ class Product:
     num_of_reviews: int
 
 
-def get_all_products() -> None:
-    pass
+driver = webdriver.Chrome()
+wait = WebDriverWait(driver, 10)
+
+def accept_cookies():
+    try:
+        cookies_button = wait.until(EC.element_to_be_clickable((By.XPATH, "//button[text()='Accept']")))
+        cookies_button.click()
+        print("Accepted cookies.")
+    except (NoSuchElementException, TimeoutException):
+        print("No cookies button found.")
+
+def scrape_page(url, paginate=False):
+    driver.get(url)
+    accept_cookies()
+
+    products = []
+
+    if paginate:
+        while True:
+            try:
+                load_more_button = wait.until(EC.element_to_be_clickable((By.CSS_SELECTOR, ".load-more-button")))
+                ActionChains(driver).move_to_element(load_more_button).click(load_more_button).perform()
+                time.sleep(1)
+            except (NoSuchElementException, TimeoutException):
+                print("No more pages to load.")
+                break
+
+    soup = BeautifulSoup(driver.page_source, "html.parser")
+    items = soup.select(".col-md-4")
+
+    for item in items:
+        title = item.select_one("a.title").text.strip() if item.select_one("a.title") else "No title"
+
+        price_text = item.select_one("h4.price").text.strip() if item.select_one("h4.price") else "0"
+        price = float(price_text.replace("$", "")) if price_text else 0.0
+
+        description = item.select_one("p.card-text").text.strip() if item.select_one("p.card-text") else "No description"
+
+        rating = len(item.select(".fa.fa-star"))
+
+        num_of_reviews_text = item.select_one(".ratings .pull-right").text.strip() if item.select_one(
+            ".ratings .pull-right") else "0 reviews"
+        num_of_reviews = int(num_of_reviews_text.split()[0]) if num_of_reviews_text else 0
```

Comment:
The conversion of num_of_reviews_text to an integer assumes that the text will always be in a valid format. Consider adding error handling to manage cases where the text cannot be converted to an integer, which could occur if the format changes or if unexpected data is encountered.